appcues/src/main/java/com/appcues/ui/RenderViewManager.kt

```diff
@@ -0,0 +1,169 @@
+                val embedView = renderContextManager.getEmbedView(renderContext) ?: return false
+                val composeView = ComposeView(context).apply { setAppcuesComposition(chromeClientView) }
+
+                embedView.addView(composeView)
+            }
+            Modal -> {
+                binding.appcuesOverlayComposeView.isVisible = true
+                binding.appcuesOverlayComposeView.setAppcuesComposition(chromeClientView)
+                // remove customers view on accessibility stack
+                setAccessibility(View.IMPORTANT_FOR_ACCESSIBILITY_NO_HIDE_DESCENDANTS)
+            }
+        }
+
+        return true
+    }
+
+    private fun ViewGroup.getOverlayLayoutBinding(): AppcuesOverlayLayoutBinding {
+        val overlayView = findViewById<ViewGroup>(R.id.appcues_overlay_layout)
+        return if (overlayView == null) {
+            // then we add
+            val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+            val binding = AppcuesOverlayLayoutBinding.inflate(inflater)
+
+            // ensures it always comes before appcues_debugger_view
+            findViewById<ViewGroup>(R.id.appcues_debugger_view)?.let {
+                addView(binding.root, indexOfChild(it))
+            } ?: run {
+                addView(binding.root)
+            }
+
+            binding.root.layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT)
+                .apply {
+                    // adds margin top and bottom according to visible status and navigation bar
+                    ViewCompat.getRootWindowInsets(this@getOverlayLayoutBinding)?.getInsets(WindowInsetsCompat.Type.systemBars())
+                        ?.let { insets ->
+                            setMargins(insets.left, insets.top, insets.right, insets.bottom)
+                        }
+                }
+
+            binding
+        } else {
+            AppcuesOverlayLayoutBinding.bind(overlayView)
+        }
+    }
+
+    private fun ComposeView.setAppcuesComposition(chromeClientView: FrameLayout) {
+        setContent {
+            AppcuesComposition(
+                viewModel = remember { viewModel!! },
+                shakeGestureListener = remember { shakeGestureListener!! },
+                logcues = scope.get(),
+                chromeClient = EmbedChromeClient(chromeClientView),
+            )
+        }
+    }
+
+    private fun onCompositionDismiss() {
+        AppcuesActivityMonitor.unsubscribe(this)
+        val parentView = AppcuesActivityMonitor.activity?.getParentView()
+
+        parentView?.post {
+            parentView.findViewTreeLifecycleOwner()?.lifecycle?.removeObserver(this@RenderViewManager)
+
+            if (renderContext == Modal) {
+                parentView.getOverlayLayoutBinding().let {
+                    it.appcuesOverlayComposeView.isVisible = false
+                    it.appcuesOverlayComposeView.setContent { }
+                }
```

Comment:
I guess I'm not yet understanding this part from the description

> the main idea is that we we will ALWAYS need to add our custom view, so we can open URLs internall in the app

Could you say more about that?